laravel/app/Http/Controllers/ScheduleDataController.php

```diff
@@ -249,109 +249,57 @@ public function getDay($day = null, $room = null, $leadSurgeon = null, $rotation
         return view('schedules.resident.schedule_table',compact('minTime', 'maxTime', 'schedule_data', 'filter_options', 'rotation_options'));
     }
 
-    public function getChoice()
-    {
-        /**
-         * Exclude Admin from selecting preferences
-         */
+    public function getChoice() {
+        // Exclude Admin from selecting preferences
         if (!Resident::where('email', $_SERVER["HTTP_EMAIL"])->exists()) {
             return view('nonpermit');
         }
         // get the id from the form
         $id = $_REQUEST['schedule_id'];
-
+        $choiceTypes = array("First", "Second", "Third");
 		// id is stored as id1_id2_id3, need to split it to get the individual ids
 		$split = explode("_", $id);
-
-		//start with the first choice
-		$choice = 1;
-		// get the first choices data
-        $schedule_data1 = ScheduleData::where('id', $split[0])->get();
-        $input[0] = array(
-            'id'=>$split[0],
-            'schedule' => ScheduleData::where('id', $split[0])->get(),
-            'choice'=>$choice,
-            'milestones'=>Milestone::where('id', $_REQUEST['milestones1'])->get(),
-            'pref_anest'=>Anesthesiologist::where('id', $_REQUEST['pref_anest1'])->get(),
-            'objectives'=>$_REQUEST['objectives1']
-        );
-
-    	// get the second choices data
-    	$choice++;
-    	// check if the second choice exists
-        if ($split[1] != 0){
-        	$schedule_data2 = ScheduleData::where('id', $split[1])->get();
-            $input[1] = array(
-                'id'=>$split[1],
-                'schedule'=>ScheduleData::where('id', $split[1])->get(),
-                'choice'=>$choice,
-                'milestones'=>Milestone::where('id', $_REQUEST['milestones2'])->get(),
-                'pref_anest'=>Anesthesiologist::where('id', $_REQUEST['pref_anest2'])->get(),
-                'objectives'=>$_REQUEST['objectives2']
-            );
-        } else {
-            $input[1] = NULL;
-        }
-
-
-    	// get the third choices data
-        $choice++;
-    	// check if the third choice exists
-        if ($split[2] != 0){
-    		$schedule_data3 = ScheduleData::where('id', $split[2])->get();
-            $input[2] = array(
-                'id'=>$split[2],
-                'schedule'=>ScheduleData::where('id', $split[2])->get(),
-                'choice'=>$choice,
-                'milestones'=>Milestone::where('id', $_REQUEST['milestones3'])->get(),
-                'pref_anest'=>Anesthesiologist::where('id', $_REQUEST['pref_anest3'])->get(),
-                'objectives'=>$_REQUEST['objectives3']
-            );
-        } else {
-            $input[2] = NULL;
+        // get current preferences
+        $currentChoices = array();
```

Comment:
style question: any reason we're using `array()` instead of `[]`? I find the latter more readable.

Reply:
Just preference, I suppose. array() is also used in other places in REMODEL - I used array() when updating the `AutoAssignment ` logic too

Reply:
okay! doesn't matter much either way imo.

Plus, at some point we should probably just run a PHP formatter and forget about it.